proc_IR_noecho.py

```diff
@@ -8,50 +8,67 @@
 fl = figlist_var()
 t2 = symbols('t2')
 logger = init_logging("info")
-#loading data in
-for searchstr,exp_type,which_exp,postproc in [
-        ('w8_200224','test_equip',2,'ab_ir2h'),
-        #('w12_200224',2),
-        #('ag_oct182019_w0_10',3),
-        #('ag_oct182019_w0_8',3),
-        #('ag_oct182019_w0_6',2),
-        #('ag_oct182019_w0_3',2),
-        #('ag_oct92019_w0_12',2),
-        #('ag_oct92019_w0_10',2),
-        #('ag_oct92019_w0_8',2),
-        #('ag_bulk_d20',2),
-        #('ag_sep062019_w0_12_IR',2),
-        #('ag_sep232019_w0_12_prod',2),
-        #('ag_sep232019_w0_8_prod',2),
-        #('ag_sep232019_w0_6_prod',2),
-        #('ag_sep232019_w0_3_prod',2),
-        #('ag_sep232019_w0_1_prod',2),
+#loading data in:
+#    this_l is the regularization lambda specific to
+#    each dataset, which must be chosen from the
+#    L-curve.
+for searchstr,exp_type,which_exp,postproc,this_l,f_range in [
+        ('w8_200224','test_equip',2,'ab_ir2h',0.008,(-150,150)),
+        #('w12_200224',2,(-150,150)),
+        #('ag_oct182019_w0_10',3,(-150,150)),
+        #('ag_oct182019_w0_8',3,(-150,150)),
+        #('ag_oct182019_w0_6',2,(-150,150)),
+        #('ag_oct182019_w0_3',2,(-150,150)),
+        #('ag_oct92019_w0_12',2,(-150,150)),
+        #('ag_oct92019_w0_10',2,(-150,150)),
+        #('ag_oct92019_w0_8',2,(-150,150)),
+        #('ag_bulk_d20',2,(-150,150)),
+        #('ag_sep062019_w0_12_IR',2,(-150,150)),
+        #('ag_sep232019_w0_12_prod',2,(-150,150)),
+        #('ag_sep232019_w0_8_prod',2,(-150,150)),
+        #('ag_sep232019_w0_6_prod',2,(-150,150)),
+        #('ag_sep232019_w0_3_prod',2,(-150,150)),
+        #('ag_sep232019_w0_1_prod',2,(-150,150)),
         ]:
     s = find_file(searchstr, exp_type=exp_type, 
             expno=which_exp, 
             postproc=postproc, lookup=postproc_dict, 
             dimname='indirect')
-    #{{{convolution and zeroth order phase correction
-    fl.next('select coherence pathway and convolve')
+    s.ift('t2')
+    #{{{ select appropriate coherence channel
     s = s['ph2',0]['ph1',-1]
-    s.convolve('t2',5)
-    fl.image(s)
     s.reorder('t2')
     ph0 = zeroth_order_ph(s['t2':0],fl=None)
     ph0 /= abs(ph0)
     s /= ph0
+    fl.next('zeroth order phase correction')
+    fl.image(s)
     #}}}
+    s.ft('t2')
     #{{{visualize phased spectra
     for j in range(ndshape(s)['indirect']):
         fl.next('Plotting phased spectra')
-        fl.plot(s['indirect',j]['t2':(-150,150)],
+        fl.plot(s['indirect',j]['t2':(0,None)],
             alpha=0.5,
             label='vd=%g'%s.getaxis('indirect')[j])
     #}}}
-    #{{{exponential curve
-    rec_curve = s['t2':(-150,150)].sum('t2')
+    #{{{exponential curve with fit
+    rec_curve = s['t2':f_range].sum('t2')
     fl.next('recovery curve')
     fl.plot(rec_curve,'o')
```

Comment:
seems to be inverted for some reason but I can't figure out why

Reply:
Inverted how, specifically?  Negative? Axis backwards?

Reply:
exponential decay rather than "recovery curve"

Reply:
this was discussed on slack + I think you have an idea of how to fix now?